tests/linen/summary_test.py

```diff
@@ -33,38 +33,65 @@ def _maybe_get_shape(x):
   return x.shape if hasattr(x, "shape") else x
 
 
-def _get_tabulate_cnn(test_sow: bool = False) -> Type[nn.Module]:
+def _get_tabulate_cnn(test_sow: bool = False, use_method: bool = False) -> Type[nn.Module]:
   @dataclasses.dataclass
   class ConvBlock(nn.Module):
```

Comment:
Can't you just have one class with both `__call__` and `block_method` implemented? (maybe with different behavior)

Then you can just make `test_sow` a class attribute as well and you don't need `_get_tabulate_cnn` anymore.

Reply:
Sounds reasonable. Will refactor.